torchrl/data/replay_buffers/samplers.py

```diff
@@ -818,28 +849,12 @@ def _get_stop_and_length(self, storage, fallback=True):
 
         else:
             try:
-                # In some cases, the storage hides the data behind "_data".
-                # In the future, this may be deprecated, and we don't want to mess
-                # with the keys provided by the user so we fall back on a proxy to
-                # the traj key.
-                if isinstance(storage, TensorStorage):
-                    try:
-                        done = storage._storage.get(self._used_end_key)
-                    except KeyError:
-                        done = storage._storage.get(("_data", self.end_key))
-                        # cache that value for future use
-                        self._used_end_key = ("_data", self.end_key)
-                    self._uses_data_prefix = (
-                        isinstance(self._used_end_key, tuple)
-                        and self._used_end_key[0] == "_data"
+                try:
+                    done = storage[:].get(self.end_key)
```

Comment:
Same here, I would suggest to use copy() or clone() instead of [:] to be more explicit.

Reply:
See comment above